src/main/java/uk/gov/pay/products/service/PaymentFinder.java

```diff
@@ -2,36 +2,64 @@
+import uk.gov.pay.products.persistence.entity.PaymentEntity;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
+import static uk.gov.pay.products.util.PaymentStatus.SUBMITTED;
+
 public class PaymentFinder {
 
     private final PaymentDao paymentDao;
     private final LinksDecorator linksDecorator;
+    private final PublicApiRestClient publicApiRestClient;
 
     @Inject
-    public PaymentFinder(PaymentDao paymentDao, LinksDecorator linksDecorator){
+    public PaymentFinder(PaymentDao paymentDao, LinksDecorator linksDecorator, PublicApiRestClient publicApiRestClient) {
         this.paymentDao = paymentDao;
         this.linksDecorator = linksDecorator;
+        this.publicApiRestClient = publicApiRestClient;
     }
 
-    @Transactional
     public Optional<Payment> findByExternalId(String paymentExternalId) {
-        return paymentDao.findByExternalId(paymentExternalId)
+        return findPaymentEntity(paymentExternalId)
                 .map(paymentEntity -> linksDecorator.decorate(paymentEntity.toPayment()));
     }
 
     @Transactional
-    public List<Payment> findByProductExternalId(String productExternalId){
-            List<Payment> payments = paymentDao.findByProductExternalId(productExternalId)
-                    .stream()
-                    .map(paymentEntity -> linksDecorator.decorate(paymentEntity.toPayment()))
-                    .collect(Collectors.toList());
-            return payments;
-        }
+    public List<Payment> findByProductExternalId(String productExternalId) {
+        List<Payment> payments = paymentDao.findByProductExternalId(productExternalId)
+                .stream()
+                .map(paymentEntity -> linksDecorator.decorate(paymentEntity.toPayment()))
+                .collect(Collectors.toList());
+        return payments;
+    }
+
+    public Optional<Payment> findByExternalIdWithExternalStatus(String paymentExternalId){
+        return  findPaymentEntity(paymentExternalId).map(paymentEntity -> {
+                Payment payment = paymentEntity.toPayment();
+                if (payment.getStatus() == SUBMITTED) {
+                    Optional<PaymentResponse> paymentResponseOptional =
+                            publicApiRestClient.getPayment(paymentEntity.getProductEntity().getPayApiToken(),
+                                    paymentEntity.getGovukPaymentId());
+                    if (paymentResponseOptional.isPresent()) {
+                        PaymentResponse paymentResponse = paymentResponseOptional.get();
+                        String status = paymentResponse.getState().getStatus();
+                        payment.setGovukStatus(status);
+                    }
+                }
+                return linksDecorator.decorate(payment);
+        });
+    }
+
+    @Transactional
+    private Optional<PaymentEntity> findPaymentEntity(String paymentExternalId){
+        return paymentDao.findByExternalId(paymentExternalId);
```

Comment:
perhaps return `Optional<Payment>` here by converting PaymentEntity.toPayment() so that Entity never goes out of the @Transactonal bound.

Reply:
Does the `@Transactional` works on private methods? Just double check with Guice docs

Reply:
Returning PaymentEntity as I need ProductEntity to use the ApiToken when querying publicapi.
You were right, it must be public